homeassistant/components/modbus/base_platform.py

```diff
@@ -185,6 +208,16 @@ def unpack_structure_result(self, registers: list[int]) -> str | None:
             v_result = []
             for entry in val:
                 v_temp = self._scale * entry + self._offset
+                if (
+                    self._min_value_threshold is not None
+                    and v_temp < self._min_value_threshold
```

Comment:
It would be a lot easier to understand if you just use min_value. something like 
```
if value < min_value 
   value = min_value
```

Reply:
min value on its own sets a value it cannot be smaller than BUT ...

This will not work for my use case ... I would have to add another setting

i.e. how do you suggest we achieve: if it is less than 0.025 mA, treat it as 0

I am open to another suggestion, as min_value on its own will not.

Reply:
BTW..  reviewing this myself, these 2 very similar bits of new code should be moved into a separate function __process_result that scales, offsets, and checks min/max

Reply:
Well you started off by suggesting only a min_value ? 

My biggest problem is that you add 2 keywords (4 in total min/max), and do not cover the cases where only 1 keywords is used (e.g. min with threshold or just threshold). This is adding a complexity to the configuration that is unwanted.

I have no problem with a simple filter function (min/max) but extending it is really something you should do with a template sensor, where you have a lot more possibilities.

It is important to keep the configuration as simple as possible, the modbus integration is responsible for communicating with modbus devices and does very simple value manipulation. Just to give you an idea we have had requests to replace "scale" with a function, allowing the value to be graduated instead of a simple scaling, the list manipulation wishes are big, and are all solved with the template sensor.

Reply:
I started with one parameter.
You asked me to add min_value and max_value which I did.
If you don't use them it will work as now.
If you just specify min_value or max_value then the pr will work as you requested.  I had to add optional thresholds to make them work for my original use case.
Please check the comments on the pr itself as they clearly indicate you can just have min_value and it will work as you expect.
This is clear in the code where the values are loaded in the constructor

Reply:
BTW.. my original proposal had been to clamp small values near 0 to 0, but what is there now is more flexible and will handle your use cases as well

Reply:
yes, and if you use a template sensor, you do not need to make a custom integration, which you need to keep updated whenever the core modbus integration changes pymodbus version.

Reply:
And not to forget my comment about what happens if you use just min_value (which will not work with this PR) or if you use just min_threshold which might have some side-effects.

Reply:
I tested min_value on own and it works as expected.
If you specify min_value_threshold on own it will do nothing. 
These cases are covered by the code